### ISSUE TITLE
How can explainer analyze just several specific layers or sub-models we chooosed?

### ISSUE THREAD

**Issue by codyuan** [shap/shap#3773]
For example, I have a model doing inference like:

    def forward(
            self,
            s,
            select_mask,
            state = None,
        ):
            s,targets=self.selector(s,select_mask)  #len(s.shape)
            logits, h = self.preprocess(s, state)
            logits = self.last(logits)
            if self.softmax_output:
                logits = F.softmax(logits, dim=-1)
            return logits, h, targets, s

So my model has 3 sub-models (selector, preprocess, last), where selector and last is MLP and preprocess is LSTM. Now I want to analyze just preprocess and last, so I call:

     shap.DeepExplainer(torch.nn.Sequential(model.preprocess,model.last), s)

However, the preprocess has two outputs and only logits will be input to last, so it got an error here.

How can I make it work? Or can I customize anohter function to replace forward when I call explainer?
